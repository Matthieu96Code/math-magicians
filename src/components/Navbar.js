import React from 'react';
import { NavLink } from 'react-router-dom';
import '../styles/Math.css';

const links = [
  { path: '/', text: 'Home' },
  { path: 'calculator', text: 'Calculator' },
  { path: 'quote', text: 'Quote' },
];

const Navbar = () => (
  <nav className="navbar">
    <ul className="menuList">
      {links.map((link) => (
        <li className="listElement" key={link.text}>
          <NavLink to={link.path} className="linkedList">{link.text}</NavLink>
        </li>
      ))}
    </ul>
  </nav>
);
export default Navbar;
