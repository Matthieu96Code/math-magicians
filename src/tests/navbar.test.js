import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import { render } from '@testing-library/react';
import Navbar from '../components/Navbar';

describe('Navbar component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

});
