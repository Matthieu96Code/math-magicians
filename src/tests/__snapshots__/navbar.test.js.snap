// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render correctly 1`] = `
<nav
  class="navbar"
>
  <ul
    class="menuList"
  >
    <li
      class="listElement"
    >
      <a
        class="linkedList"
        href="/home"
      >
        Home
      </a>
    </li>
    <li
      class="listElement"
    >
      <a
        class="linkedList"
        href="/calculator"
      >
        Calculator
      </a>
    </li>
    <li
      class="listElement"
    >
      <a
        class="linkedList"
        href="/quote"
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
